### PR TITLE
Refactor and test ballcamera

### PR DESCRIPTION
**Pull request by josephhadad05**
The pull request creates an inner class within the BallCamera class used to get a Trajectory from a specific ball translation. I was able to break everything down into smaller methods and move it into the `getTrajectory` method within the subclass. Along with that I created the two unit-tests requested, both of which passed.

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/BallCamera.java [hunk 1]
```diff
@@ -11,7 +11,7 @@
 import edu.wpi.first.wpilibj.DriverStation;
 import frc.robot.Constants;
 
-public abstract class BallCamera extends AbstractCamera {
+public class BallCamera extends AbstractCamera {
```
Comment (paulelliott) [inline src/main/java/frc/robot/subsystems/BallCamera.java:14]: Good catch! Classes that get instantiated should not be abstract.

**Comment by paulelliott** [comment]
This looks great! The next step is to write a few tests with specific inputs and assert that specific outputs are generated. You can tackle that in this PR or open a subsequent one for it. Up to you.

**Comment by josephhadad05** [comment]
Alright, thanks. I will go ahead create some more unit tests here.

**Comment by josephhadad05** [comment]
I was able to add two additional checks to the program, both of which passed.

**Pull request merged**